cmd/contour/serve.go

```diff
@@ -275,6 +278,28 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 		FieldLogger: log.WithField("context", "endpointstranslator"),
 	}
 
+	// snapshotCache is used to store the state of what all xDS services should
+	// contain at any given point in time.
+	snapshotCache := cache.NewSnapshotCache(false, cache.IDHash{},
+		log.WithField("context", "XDSServer"))
+
+	// snapshotHandler is used to produce new snapshots when the internal state changes
+	// for any xDS resource.
+	snapshotHandler := &contour.SnapshotHandler{
+		CacheHandler:        ch,
+		EndpointsTranslator: et,
+		SnapshotCache:       snapshotCache,
+	}
+
+	// Pass the snapshotHandler off to the CacheHandler & EndpointsTranslator
+	// so they are able to inform the snapshotHandler that the xDS resources
+	// have changed and should be updated in Envoy.
+	ch.SnapshotHandler = snapshotHandler
+	et.SnapshotHandler = snapshotHandler
+
+	// Trigger an initial snapshot for static resources
```

Comment:
Can you explain this in more detail? What static resources are getting snapshotted? Don't we want to wait until the informer caches sync before sending anything to an Envoy?

Reply:
There are static resources in listeners that need to get set for probes and other bits to come up. If you don't do this, then you'd have to wait for some resource to show up and trigger a dag rebuild to get these pushed into Envoy.

Reply:
We ought to keep the previous semantics of not pushing and xDS resources until after the informer caches sync (preferably IMHO until after the first DAG rebuild), so that we don't remove working config and then replace it

Reply:
Yeah that's a good point. We really need to wait until the first dag build though like you said, the caches being sync'd doesn't mean that the dag is built and ready. 

I can look to add that in here or a followup PR.